### PR TITLE
chore: insert colums more the the table shouldn't panic

### LINKED ISSUE #7434: panic: insert colums more the the table
**Summary**

```
create table st1(a string);
insert into st1 values ('a', 'b');
```

```
2022-09-01T10:05:31.374490Z ERROR common_tracing::panic_hook: panicked at 'index out of bounds: the len is 1 but the index is 1', src/query/datavalues/src/data_schema.rs:84:10 backtrace=Backtrace [{ fn: "std::backtrace::Backtrace::create" }, { fn: "std::backtrace::Backtrace::force_capture" }, { fn: "common_tracing::panic_hook::log_panic" }, { fn: "std::panicking::rust_panic_with_hook" }, { fn: "std::panicking::begin_panic_handler::{{closure}}" }, { fn: "std::sys_common::backtrace::__rust_end_short_backtrace" }, { fn: "_rust_begin_unwind" }, { fn: "core::panicking::panic_fmt" }, { fn: "core::panicking::panic_bounds_check" }, { fn: "databend_query::sql::planner::binder::insert::ValueSourceV2::read::{{closure}}" }, { fn: "<core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll" }, { fn: "<core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll" }, { fn: "databend_query::sql::planner::binder::Binder::bind_statement::{{closure}}" }, { fn: "<core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll" }, { fn: "<core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll" }, { fn: "<core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll" }, { fn: "<core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll" }, { fn: "<core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll" }, { fn: "opensrv_mysql::AsyncMysqlIntermediary<B,S,W>::run_with_options::{{closure}}" }, { fn: "<core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll" }, { fn: "std::panicking::try" }, { fn: "tokio::runtime::task::harness::Harness<T,S>::poll" }, { fn: "tokio::runtime::thread_pool::worker::Context::run_task" }, { fn: "tokio::runtime::thread_pool::worker::run" }, { fn: "tokio::runtime::task::raw::poll" }, { fn: "std::sys_common::backtrace::__rust_begin_short_backtrace" }, { fn: "core::ops::function::FnOnce::call_once{{vtable.shim}}" }, { fn: "std::sys::unix::thread::Thread::new::thread_start" }, { fn: "__pthread_start" }] panic.file="src/query/datavalues/src/data_schema.rs" panic.line=84 panic.column=10
```

Panic boom...

### PR DESCRIPTION
**Pull request by xudong963**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

Fixes #7434

### CODE REVIEW

**Comment by BohuTANG** [comment]
Need a logical test for it:)

**Review by sundy-li — APPROVED** [review]

**Review by BohuTANG — APPROVED** [review]

File: tests/logictest/suites/base/03_dml/03_0016_insert_into_values [hunk 1]
```diff
@@ -47,5 +47,13 @@ a"b"c'd
 a"b"c\'d
 
 statement ok
-DROP DATABASE db1;
+create table if not exists st1(a string);
+
+statement error 1105
```
Comment (xudong963) [inline tests/logictest/suites/base/03_dml/03_0016_insert_into_values:52]:
```suggestion
statement error 1015
```

**Comment by xudong963** [comment]
https://github.com/datafuselabs/databend/runs/8150725760?check_suite_focus=true cc @drmingdrmer

**Comment by drmingdrmer** [comment]
> ```
> failures:
>     grpc::metasrv_grpc_schema_api::test_meta_grpc_client_cluster
>     grpc::metasrv_grpc_schema_api_follower_follower::test_meta_grpc_client_database_create_get_drop
> 
> test result: FAILED. 49 passed; 2 failed; 0 ignored; 0 measured; 0 filtered out; finished in 75.59s
> 
>      Running tests/it/main.rs (target/debug/deps/it-527ec1ecdc50d50e)
> ```
> 
> https://github.com/datafuselabs/databend/runs/8150725760?check_suite_focus=true cc @drmingdrmer

Artifact uploading failed due to log file path changes. I can not see what's happening in this round.

I'll fix up the artifact path and extend the timeout for the next time to see what's happened.

**Pull request merged**